### DISCUSSION TITLE
Two pull requests for copilot.vim

### DISCUSSION THREAD

**Discussion by petronny** [community/community#34855] [Copilot Conversations] [answered]
Hi I have two pull requests for copilot.vim but I'm not a collaborator.

The first one converts the port in the proxy setting to integer or else the agent will complain.
Commit: https://github.com/petronny/copilot.vim/commit/95b9f5d2100235a741f4ef8b5556d4691617b4c8
Link to create PR: https://github.com/petronny/copilot.vim/pull/new/fix_proxy_port

The second one increases the priority of `g:copilot_agent_command` so we don't need to delete the existing `agent.js` to let it works.
Commit: https://github.com/petronny/copilot.vim/commit/f0f032916b650ac563c3518c3e722316a1170e34
Link to create PR: https://github.com/petronny/copilot.vim/pull/new/fix_copilot_agent_command

Ping @tpope .

**Comment by tpope** [comment, marked as answer]
A bit miffed at the proxy one, I set up a whole proxy server just to test that. I'll definitely get that fixed.

`g:copilot_agent_command` is a development backdoor. I'm a little unclear why you would want to use it as an end user?

**Reply by petronny** [reply]
Oh, I'm thinking about writing custom agent for other completion backends.

Currently I use a shell wrapper to agent.js to capture its inputs and outputs.
